x-pack/plugins/security_solution/public/common/hooks/endpoint/use_alert_response_actions_support.ts

```diff
@@ -0,0 +1,241 @@
+    agentIdField: string;
+  };
+}
+
+type AlertAgentActionsSupported = Record<ResponseActionsApiCommandNames, boolean>;
+
+/**
+ * Determines the level of support that an alert's host has for Response Actions.
+ * This hook already checks feature flags to determine the level of support that we have available
+ */
+export const useAlertResponseActionsSupport = (
+  eventData: TimelineEventsDetailsItem[] | null = []
+): AlertResponseActionsSupport => {
+  const isAlert = useMemo(() => {
+    return some({ category: 'kibana', field: 'kibana.alert.rule.uuid' }, eventData);
+  }, [eventData]);
+
+  const agentType: ResponseActionAgentType | undefined = useMemo(() => {
+    if ((find({ field: 'agent.type' }, eventData)?.values ?? []).includes('endpoint')) {
+      return 'endpoint';
+    }
+
+    const eventModuleValues = find({ field: 'event.module' }, eventData)?.values ?? [];
+
+    if (eventModuleValues.includes('sentinel_one')) {
+      return 'sentinel_one';
+    }
+
+    if (eventModuleValues.includes('crowdstrike')) {
+      return 'crowdstrike';
+    }
+
+    return undefined;
+  }, [eventData]);
+
+  const isFeatureEnabled: boolean = useMemo(() => {
+    return agentType ? isAgentTypeAndActionSupported(agentType) : false;
+  }, [agentType]);
+
+  const agentId: string = useMemo(() => {
+    if (!agentType) {
+      return '';
+    }
+
+    if (agentType === 'endpoint') {
+      return getAlertDetailsFieldValue({ category: 'agent', field: 'agent.id' }, eventData);
+    }
+
+    if (agentType === 'sentinel_one') {
+      return getAlertDetailsFieldValue(
+        { category: 'observer', field: RESPONSE_ACTIONS_ALERT_AGENT_ID_FIELD.sentinel_one },
+        eventData
+      );
+    }
+
+    if (agentType === 'crowdstrike') {
+      return getAlertDetailsFieldValue(
+        { category: 'crowdstrike', field: RESPONSE_ACTIONS_ALERT_AGENT_ID_FIELD.crowdstrike },
+        eventData
+      );
+    }
+
+    return '';
+  }, [agentType, eventData]);
+
+  const doesHostSupportResponseActions = useMemo(() => {
+    return Boolean(isFeatureEnabled && isAlert && agentId && agentType);
+  }, [agentId, agentType, isAlert, isFeatureEnabled]);
+
+  const agentIdField = useMemo(() => {
+    if (agentType) {
+      return RESPONSE_ACTIONS_ALERT_AGENT_ID_FIELD[agentType];
+    }
+
+    return '';
+  }, [agentType]);
+
+  const supportedActions = useMemo(() => {
+    return RESPONSE_ACTION_API_COMMANDS_NAMES.reduce<AlertAgentActionsSupported>(
+      (acc, responseActionName) => {
+        acc[responseActionName] = false;
+
+        if (agentType && isFeatureEnabled) {
+          acc[responseActionName] = isAgentTypeAndActionSupported(
+            agentType,
+            responseActionName,
+            'manual'
+          );
+        }
+
+        return acc;
+      },
+      {} as AlertAgentActionsSupported
+    );
+  }, [agentType, isFeatureEnabled]);
+
+  const hostName = useMemo(() => {
+    // TODO:PT need to check if crowdstrike event has `host.name`
```

Comment:
👍  I doubled checked, and I believe we can assume that it does contain `host.name` because mappings says so. Some events I have been testing do not have them though, but I believe it's because the original (crowdstrike) event didn't have it either.